src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IForEachLoopStatement.cs

```diff
@@ -788,8 +846,178 @@ static void Main()
+                Diagnostic(ErrorCode.ERR_InExpected, ",").WithLocation(6, 19),
+                // CS0230: Type and identifier are both required in a foreach statement
+                //         foreach (i, j in x)
+                Diagnostic(ErrorCode.ERR_BadForeachDecl, ",").WithLocation(6, 19),
+                // CS1525: Invalid expression term ','
+                //         foreach (i, j in x)
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, ",").WithArguments(",").WithLocation(6, 19),
+                // CS1026: ) expected
+                //         foreach (i, j in x)
+                Diagnostic(ErrorCode.ERR_CloseParenExpected, ",").WithLocation(6, 19),
+                // CS1525: Invalid expression term ','
+                //         foreach (i, j in x)
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, ",").WithArguments(",").WithLocation(6, 19),
+                // CS1002: ; expected
+                //         foreach (i, j in x)
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, ",").WithLocation(6, 19),
+                // CS1513: } expected
+                //         foreach (i, j in x)
+                Diagnostic(ErrorCode.ERR_RbraceExpected, ",").WithLocation(6, 19),
+                // CS1002: ; expected
+                //         foreach (i, j in x)
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "in").WithLocation(6, 23),
+                // CS1513: } expected
+                //         foreach (i, j in x)
+                Diagnostic(ErrorCode.ERR_RbraceExpected, "in").WithLocation(6, 23),
+                // CS1002: ; expected
+                //         foreach (i, j in x)
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, ")").WithLocation(6, 27),
+                // CS1513: } expected
+                //         foreach (i, j in x)
+                Diagnostic(ErrorCode.ERR_RbraceExpected, ")").WithLocation(6, 27),
+                // CS0103: The name 'j' does not exist in the current context
+                //         foreach (i, j in x)
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "j").WithArguments("j").WithLocation(6, 21)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<BlockSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact, WorkItem(17602, "https://github.com/dotnet/roslyn/issues/17602")]
+        public void IForEachLoopStatement_WithInvalidLoopControlVariable_02()
+        {
+            string source = @"
+class X
+{
+    public static void M(int[] x)
+    /*<bind>*/{
+        foreach (x[0] in x)
+        {
+        }
+    }/*</bind>*/
+}
+";
+            string expectedOperationTree = @"
+IBlockStatement (1 statements) (OperationKind.BlockStatement, IsInvalid) (Syntax: '{ ... }')
+  IForEachLoopStatement (LoopKind.ForEach) (OperationKind.LoopStatement, IsInvalid) (Syntax: 'foreach (x[ ... }')
+    LoopControlVariable: null
```

Comment:
> LoopControlVariable: null [](start = 3, length = 26)

It looks like we are dropping the expression from the tree. This is likely to require changes in bound nodes, please open an issue to follow up.

Reply:
Filed https://github.com/dotnet/roslyn/issues/22103